curp/src/server/raw_curp/mod.rs

```diff
@@ -775,13 +797,14 @@ impl<C: 'static + Command> RawCurp<C> {
         for cmd in recovered_cmds {
             let _ig_sync = cb_w.sync.insert(cmd.id().clone()); // may have been inserted before
             let _ig_spec = sp_l.insert(Arc::clone(&cmd)); // may have been inserted before
-            let index = log.push_cmd(term, Arc::clone(&cmd));
-            debug!(
-                "{} recovers speculatively executed cmd({}) in log[{}]",
-                self.id(),
-                cmd.id(),
-                index,
-            );
+            if let Ok(index) = log.push_cmd(term, Arc::clone(&cmd)) {
+                debug!(
+                    "{} recovers speculatively executed cmd({}) in log[{}]",
+                    self.id(),
+                    cmd.id(),
+                    index,
+                );
+            }
```

Comment:
Shall we use `expect` on `push_cmd` if it returns false? It seems like an intolerable fault if the cmds can't be recovered correctly.

Reply:
Fixed